lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -2103,70 +2105,275 @@ static std::string to_string(VersionNumber Version) {
   return to_string(static_cast<uint32_t>(Version));
 }
 
-std::istream &operator>>(std::istream &I, SPIRVModule &M) {
-  SPIRVDecoder Decoder(I, M);
-  SPIRVModuleImpl &MI = *static_cast<SPIRVModuleImpl *>(&M);
-  // Disable automatic capability filling.
+namespace {
+SPIRVEntry *parseAndCreateSPIRVEntry(SPIRVWord &WordCount, Op &OpCode,
```

Comment:
Will it be a good idea to split this into 'parseEntry and createEntry?

Reply:
Yes and it should be considered one of the next steps. Unfortunately, basing on my today (02/09/24) evaluation this function requires much more work to make full split possible. It's worth noting that the function (it's origin getEntry) is also one of the major contributor to overall slowness of the translator - it's responsible for ~40% of overall runtime (SPIR-V->IR) based on my recent benchmark perf measurement.

Reply:
Please add a TODO above the function definition with this info. Thanks